server/src/main/java/org/elasticsearch/index/query/SearchExecutionContext.java

```diff
@@ -681,7 +681,10 @@ public IndexReader getIndexReader() {
     }
 
     /** Return the current {@link IndexSearcher}, or {@code null} if no index reader is available,
-     *  for instance if this rewrite context is used to index queries (percolation). */
+     * which happens in two cases:
+     * 1. if this rewrite context is used to index queries (percolation)
+     * 2. if we use mapping information to skip shards while doing shards pre-filtering in the 'can match' phase
```

Comment:
Basically when a `SearchExecutionContext` is used with no indexReader.

Reply:
is the second still true given the recent improvements you made? Are there places where we do mappings based can match with a null searcher?